flytekit/core/workflow.py

```diff
@@ -135,98 +163,114 @@ def name(self) -> str:
 
     @property
     def interface(self) -> _interface_models.TypedInterface:
         return self._interface
 
     @property
-    def workflow_metadata(self) -> Optional[WorkflowMetadata]:
-        return self._workflow_metadata
+    def output_bindings(self) -> List[_literal_models.Binding]:
+        return self._output_bindings
 
     @property
-    def workflow_metadata_defaults(self):
-        return self._workflow_metadata_defaults
+    def nodes(self) -> List[Node]:
+        return self._nodes
+
+    def __repr__(self):
+        return (
+            f"WorkflowBase - {self._name} && "
+            f"Inputs ({len(self._python_interface.inputs)}): {self._python_interface.inputs} && "
+            f"Outputs ({len(self._python_interface.outputs)}): {self._python_interface.outputs} && "
+            f"Output bindings: {self._output_bindings} && "
+        )
 
-    def task_name(self, t: PythonAutoContainerTask) -> str:
-        return f"{self.name}.{t.__module__}.{t.name}"
+    def __call__(self, *args, **kwargs):
+        if len(args) > 0:
+            raise AssertionError("Only Keyword Arguments are supported for Workflow executions")
 
-    def compile(self, **kwargs):
-        """
-        Supply static Python native values in the kwargs if you want them to be used in the compilation. This mimics
-        a 'closure' in the traditional sense of the word.
-        """
         ctx = FlyteContext.current_context()
-        self._input_parameters = transform_inputs_to_parameters(ctx, self._native_interface)
-        all_nodes = []
-        prefix = f"{ctx.compilation_state.prefix}-{self.short_name}-" if ctx.compilation_state is not None else None
-        with ctx.new_compilation_context(prefix=prefix, task_resolver=self) as comp_ctx:
-            # Construct the default input promise bindings, but then override with the provided inputs, if any
-            input_kwargs = construct_input_promises([k for k in self.interface.inputs.keys()])
+
+        # The first condition is compilation.
+        if ctx.compilation_state is not None:
+            input_kwargs = self.python_interface.default_inputs_as_kwargs
             input_kwargs.update(kwargs)
-            workflow_outputs = self._workflow_function(**input_kwargs)
-            all_nodes.extend(comp_ctx.compilation_state.nodes)
+            return create_and_link_node(ctx, entity=self, interface=self.python_interface, **input_kwargs)
 
-            # This needs a longer comment.
-            for n in comp_ctx.compilation_state.nodes:
-                if isinstance(n.flyte_entity, PythonAutoContainerTask) and n.flyte_entity.task_resolver == self:
-                    logger.debug(f"WF {self.name} saving task {n.flyte_entity.name}")
-                    self.add(n.flyte_entity)
+        # Next is executing as a subworkflow
+        elif (
+            ctx.execution_state is not None and ctx.execution_state.mode == ExecutionState.Mode.LOCAL_WORKFLOW_EXECUTION
```

Comment:
is this true? it's executing as a subworkflow for LOCAL_WORKFLOW_EXECUTION

Reply:
yeah. will clarify.